azure-functions-java-core/src/main/java/com/microsoft/azure/serverless/functions/annotation/HttpTrigger.java

```diff
@@ -11,18 +11,108 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * <p>The HttpTrigger annotation is applied to Azure functions that will be triggered by a call to the HTTP endpoint that
+ * the function is located at. The HttpTrigger annotation should be applied to a method parameter of type
+ * {@link com.microsoft.azure.serverless.functions.HttpRequestMessage}. For example:</p>
```

Comment:
The HttpTrigger annotation could be applied to a method parameter of various types, such as int, String, Optional&lt;String&gt;, any POJO types and HttpRequestMessage&lt;T&gt;.

Reply:
Thanks for clarifying. Is there any documentation that you know of that clarifies this in more detail - when should a developer use one type over another, etc?